tests/tensor_search/test_multimodal_tensor_combination.py

```diff
@@ -0,0 +1,882 @@
+                                                                 index_name=self.index_name_1, document_id="0",
+                                                                 show_vectors=True)['_tensor_facets'][0]["_embedding"])
+        text_tensor = \
+            np.array(tensor_search.get_document_by_id(config=self.config, index_name=self.index_name_1, document_id="1",
+                                                      show_vectors=True)['_tensor_facets'][0]["_embedding"])
+        image_tensor = \
+            np.array(tensor_search.get_document_by_id(config=self.config, index_name=self.index_name_1, document_id="2",
+                                                      show_vectors=True)['_tensor_facets'][0]["_embedding"])
+
+        assert np.sum(combo_tensor - (text_tensor * -1 + image_tensor * 1)) == 0
+
+    def test_multimodal_tensor_combination_zero_weight(self):
+        def get_score(document):
+            try:
+                tensor_search.delete_index(config=self.config, index_name=self.index_name_1)
+            except IndexNotFoundError as e:
+                pass
+
+            tensor_search.create_vector_index(
+                index_name=self.index_name_1, config=self.config, index_settings={
+                    IndexSettingsField.index_defaults: {
+                        IndexSettingsField.model: "ViT-B/32",
+                        IndexSettingsField.treat_urls_and_pointers_as_images: True,
+                        IndexSettingsField.normalize_embeddings: False
+                    }
+                })
+
+            tensor_search.add_documents(config=self.config, index_name=self.index_name_1, docs=[document],
+                                        auto_refresh=True, mappings = {"combo_text_image" : {"type":"multimodal_combination",
+            "weights":{"image_field": 0,"text_field": 1}}})
+
+            self.assertEqual(1, tensor_search.get_stats(config=self.config, index_name=self.index_name_1)[
+                "numberOfDocuments"])
+            res = tensor_search.search(config=self.config, index_name=self.index_name_1,
+                                       text="", result_count=1)
+
+            return res["hits"][0]["_score"]
+
+        score_1 = get_score({
+            "text_field": "A rider is riding a horse jumping over the barrier.",
+            # "image_field": "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg",
+        })
+
+        score_3 = get_score({
+            "combo_text_image": {
+            "text_field" : "A rider is riding a horse jumping over the barrier.",
+            "image_field" : "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg",
+                },
+        })
+
+        self.assertEqual(score_1, score_3)
+
+    def test_multimodal_tensor_combination_vectorise_call(self):
+        """check if the chunks are properly created in the add_documents"""
```

Comment:
Can we also query OpenSearch, to see if the documents are created as expected?

Reply:
done